packages/salesforcedx-vscode-core/src/commands/functions/forceFunctionStart/ForceFunctionContainerlessStartExecutor.ts

```diff
@@ -39,12 +40,16 @@ export class ForceFunctionContainerlessStartExecutor extends ForceFunctionStartE
     channelService.appendLine(
       `Starting ${functionName} of type ${functionLanguage}`
     );
+
     const localRun = new LocalRun(functionLanguage, {
       path: functionDirPath,
       port: FUNCTION_DEFAULT_PORT,
       debugPort: FUNCTION_DEFAULT_DEBUG_PORT
     });
 
+    const debugType = functionLanguage === functionType.JAVA ? 'java' : 'node';
+    FunctionService.instance.updateFunction(functionDirPath, debugType);
```

Comment:
So in the container function flow this happens in an odd place so I missed it for containerless function.  This updates the debug type after the function is started (which is required to run the service method for getting the type.